File: src/equiv/TrailingZeroes2.ml
module JB = Javalib_pack.JBasics
module QID = QualifiedIdentity
module E = Expr

let _ =
  let x = E.Var (Var.Mk (QID.QID ["x"], Type.Int)) in

  Equiv.run "TrailingZeroes2"
            [(JB.TBasic `Int)]
            (JB.TBasic `Int)
            "trailingZeroes2"
            "trailingZeroes1"
            [x]
            [E.mk_ige x (E.Int 5)]

File: benchmark/equiv/TrailingZeroes2.java
public class TrailingZeroes2 {
  static public int trailingZeroes1(int n) {
    int num;
    for (num = 0; n != 0; n /= 5, num += n)
      ;
    return num;
  }

  static public int trailingZeroes2(int n) {
    int sum = 0;
    while (n >= 5) {
      sum += n / 5;
      n = n / 5;
    }
    return sum;
  }
}
